File: accepted/58-length-of-last-word.py
# https://leetcode.com/problems/length-of-last-word/
class Solution:
    # @param {string} s
    # @return {integer}
    def lengthOfLastWord(self, s):
        ret = 0
        cur = 0
        
        for c in s:
            cur = 0 if ' ' == c else (cur + 1)
            ret = cur if 0 != cur else ret
        
        return ret

File: accepted/58-length-of-last-word.c
/* https://leetcode.com/problems/length-of-last-word/ */
int lengthOfLastWord(char* s) {
    int ret = 0, cur = 0;
    while ('\0' != *s) {
        cur = (' ' == *s) ? 0 : cur + 1;
        ret = (0 == cur) ? ret : cur;
        s++;
    }
    return ret;
}
